### PR TITLE
SideNavigation: new component (V1)

### PR DESCRIPTION
**Pull request by AlbertCarreras**
#### What changed?

SideNavigation: new component

#### Why?

V1 in alpha release. More changes coming to the component including nested navigation items and an improved keyboard navigation 

### Links

- [Figma](https://www.figma.com/file/kK9TSF41utPLF2CIRoGLhX/Gestalt-SideNav-Component?node-id=285%3A4844)
- [TDD](https://paper.dropbox.com/doc/TDD-SideNavigation--BkRWqAZLEiLaYaoK1dA32uFvAg-4IWmy1gXM95tF7SNNKh2A)

### Checklist

- [x] Added unit and Flow Tests
- [x] Added documentation + accessibility tests
- [x] Verified accessibility: keyboard & screen reader interaction
- [x] Checked dark mode, responsiveness, and right-to-left support
- [x] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/pages/sidenavigation.js [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+      <SideNavigation
+        accessibilityLabel="Header example"
+        header={
+          <RadioGroup legend="Sort by?" id="example-header">
+            <Flex gap={2}>
+              <RadioGroup.RadioButton
+                checked={organisedBy === 'categorized'}
+                id="category"
+                label="Category"
+                name="sidebarSort"
```
Comment (ayeshakmaz) [inline docs/pages/sidenavigation.js:262]: these are showing as 3 of 4 and 4 of 4, make sure the name is unique

File: packages/gestalt/src/SideNavigationItem.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  /**
+   * When set to 'page' or 'section', it displays the item in "active" state. See the [Accessibility](https://gestalt.pinterest.systems/SideNavigation#Accessibility) guidelines to learn more.
+   */
+  active?: 'page' | 'section',
+  /**
+   * When supplied, will display a [Badge](https://gestalt.pinterest.systems/badge) next to the item's label. See the [Badges](https://gestalt.pinterest.systems/SideNavigation#Badge) variant to learn more.
+   */
+  badge?: {|
+    text: string,
+    type?: 'info' | 'error' | 'warning' | 'success' | 'neutral' | 'darkWash' | 'lightWash',
```
Comment (ayeshakmaz) [inline packages/gestalt/src/SideNavigationItem.js:23]: I think we should remove light wash and dark wash, they're only intended for use over images, and that shouldn't happen here

File: packages/gestalt/src/SideNavigationItem.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        onMouseEnter={() => setHovered(true)}
+        onMouseLeave={() => setHovered(false)}
+        onFocus={() => setHovered(true)}
+        onBlur={() => setHovered(false)}
+        role="link"
+        rounding={2}
+        tapStyle="compress"
+        onTap={({ event }) => onSelect?.({ event, item })}
+      >
+        <Box color={itemColor} paddingX={4} height={44} maxHeight={44} rounding={2}>
```
Comment (ayeshakmaz) [inline packages/gestalt/src/SideNavigationItem.js:98]: sounds like we're getting rid of maxHeight here

File: packages/gestalt/src/SideNavigationItem.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+                        role="status"
+                      />
+                    ) : null}
+                  </Box>
+                )}
+              </Text>
+            </Flex.Item>
+            {counter && (
+              <Flex.Item flex="grow" alignSelf="center">
+                <Box display="visuallyHidden">{`, `}</Box>
```
Comment (ayeshakmaz) [inline packages/gestalt/src/SideNavigationItem.js:140]: Really like these

**Review by ayeshakmaz — APPROVED** [review]

File: docs/pages/sidenavigation.js [hunk 5]
```diff
@@ -0,0 +1,638 @@
...
+          />
+        }
+      />
+
+      <GeneratedPropTable generatedDocGen={generatedDocGen.SideNavigation} />
+
+      <AccessibilitySection name={generatedDocGen.SideNavigation?.displayName}>
+        <MainSection.Subsection
+          title="Active item"
+          description='Sidenavigation.Item has an "active" state that visually identifies it. To set Sidenavigation.Item to active pass "page" (page redirect) or "section" (selected content section). Use routing hooks from React.Router or other frameworks to identify the current route. For example, if the current pathname matches the Sidenavigation.Item href, set Sidenavigation.Item to "page".'
```
Comment (ayeshakmaz) [inline docs/pages/sidenavigation.js:75]: the SideNavigation.Item is the only doc fix I'd recommend, hard to read when navigation isn't capitalized

File: docs/pages/sidenavigation.js [hunk 6]
```diff
@@ -0,0 +1,638 @@
...
+    item={{ label: 'Tune your home feed', value: 'tune_homefeed' }}
+    badge={{ text: 'New', type: 'info' }}
+  />
+</SideNavigation>
+        `}
+        slimBanner={
+          <SlimBanner
+            type="warning"
+            iconAccessibilityLabel="Warning"
+            message="This component is on alpha phase, still under developoment. The component will support three nested levels and accessibility keyboard navigation. The component will change behavior and the API might also change in future component version releases."
```
Comment (ayeshakmaz) [inline docs/pages/sidenavigation.js:65]: "This component is in the alpha phase and is still under development."

File: docs/pages/sidenavigation.js [hunk 7]
```diff
@@ -0,0 +1,638 @@
...
+        >
+          <MainSection.Card
+            shaded
+            defaultCode={`
+function Example() {
+  const reactRouterPath = '/sidenavigation'
+
+  return (
+    <Box height={300} overflow="scroll">
+      <SideNavigation accessibilityLabel="Main example">
```
Comment (ayeshakmaz) [inline docs/pages/sidenavigation.js:85]: I think your test failure will be fixed if you change this label (duplicated from top example)

File: docs/components/COMPONENT_DATA.js [hunk 8]
```diff
@@ -920,6 +920,29 @@ const GENERAL_COMPONENT_LIST: ListItemType = [
       responsive: 'ready',
     },
   },
+  {
+    svg: <SegmentedControl />,
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply docs/components/COMPONENT_DATA.js:924]: DONE.

File: docs/pages/sidenavigation.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/pages/sidenavigation.js:262]: DONE

File: packages/gestalt/src/SideNavigationItem.js (hunk 2 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/SideNavigationItem.js:23]: DONE

File: packages/gestalt/src/SideNavigationItem.js (hunk 3 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/SideNavigationItem.js:98]: to minHEight here

File: docs/pages/sidenavigation.js (hunk 6 above)
Comment (AlbertCarreras) [reply docs/pages/sidenavigation.js:65]: DONE

File: docs/pages/sidenavigation.js (hunk 7 above)
Comment (AlbertCarreras) [reply docs/pages/sidenavigation.js]: DONE

File: docs/pages/sidenavigation.js (hunk 6 above)
Comment (AlbertCarreras) [reply docs/pages/sidenavigation.js]: DONE

**Pull request merged**